src/base64_ops.rs

```diff
@@ -0,0 +1,94 @@
+use crate::allocator::{Allocator, Atom, NodePtr};
+use crate::cost::check_cost;
+use crate::cost::Cost;
+use crate::err_utils::err;
+use crate::op_utils::{atom_len, get_args, new_atom_and_cost, MALLOC_COST_PER_BYTE};
+use crate::reduction::{EvalErr, Reduction, Response};
+use crate::SExp;
+use base64ct::{Base64UrlUnpadded, Encoder, Encoding};
+
+const BASE64_ENCODE_BASE_COST: Cost = 40;
+const BASE64_COST_PER_ARG: Cost = 130;
+const BASE64_DECODE_BASE_COST: Cost = 400;
+const BASE64_COST_PER_BYTE: Cost = 3;
+
+// this computes the size of the resulting base64 encoded string, given a number
+// of input bytes. We pre-allocate the string.
+const fn encoded_len(n: usize) -> Option<usize> {
```

Comment:
Does this need to be `const`?

Reply:
I don't think there are any use cases for it. I copied it from `base64ct` and it's `const` there. It's needed to pre-allocate the right size output buffer.